logstash-core/lib/logstash/modules/logstash_config.rb

```diff
@@ -0,0 +1,46 @@
+# encoding: utf-8
+require "logstash/namespace"
+require_relative "file_reader"
+
+module LogStash module Modules class LogStashConfig
+
+  # We name it `modul` here because `module` has meaning in Ruby.
+  def initialize(modul, settings)
+    @directory = ::File.join(modul.directory, "logstash")
+    @name = modul.module_name
+    @settings = settings
+  end
+
+  def template
+    ::File.join(@directory, "#{@name}.conf.erb")
+  end
+
+  def setting(value, default)
+    @settings.fetch(value, default)
+  end
+
+  def elasticsearch_output_config
+    hosts = setting("var.output.elasticsearch.hosts", "localhost:9200").split(',').map do |s|
+      '"' + s.strip + '"'
+    end.join(',')
+    index = "#{@name}-#{setting("var.output.elasticsearch.index_suffix", "%{+YYYY.MM.dd}")}"
+    password = "#{setting("var.output.elasticsearch.password", "changeme")}"
+    user = "#{setting("var.output.elasticsearch.user", "elastic")}"
+    <<-CONF
+elasticsearch {
+hosts => [#{hosts}]
+index => "#{index}"
+password => "#{password}"
+user => "#{user}"
+manage_template => false
+}
+CONF
+  end
```

Comment:
But on the other hand, sending to ES is the most common use case, so giving people an easy macro in their template might be a good idea.